compiler/semtypes.nim

```diff
@@ -1454,7 +1454,7 @@ proc semGeneric(c: PContext, n: PNode, s: PSym, prev: PType): PType =
     recomputeFieldPositions(tx, tx.n, position)
 
 proc maybeAliasType(c: PContext; typeExpr, prev: PType): PType =
-  if typeExpr.kind in {tyObject, tyEnum, tyDistinct, tyForward} and prev != nil:
+  if typeExpr.kind in {tyObject, tyEnum, tyDistinct, tyForward, tyGenericBody} and prev != nil:
```

Comment:
Same question.

Reply:
also related to the bugfixes: without this:

```
ttypetraits.nim(133, 28) Error: type mismatch: got <FooInst, Foo2Inst>
but expected one of:
proc `==`[T: tuple |
    object](x, y: T): bool
  first type mismatch at position: 2
  required type for y: T: tuple or object
  but expression 'default(Foo2Inst)' is of type: Foo2Inst
```